GetServicePreferences/handler.ts

```diff
@@ -128,29 +130,35 @@ const getUserServicePreferencesOrDefault = (
           pref => toUserServicePreferenceFromModel(pref)
         )
       )
-    )
+    ),
+    TE.map(_ => ({
+      serviceCategory,
+      servicePreferences: _
+    }))
   );
 
 /**
  * Return a type safe GetServicePreferences handler.
  */
 export const GetServicePreferencesHandler = (
-  profileModels: ProfileModel,
-  servicePreferencesModel: ServicesPreferencesModel
+  profileModel: ProfileModel,
+  serviceModel: ServiceModel,
+  servicePreferencesModel: ServicesPreferencesModel,
+  activationModel: ActivationModel
   // eslint-disable-next-line arrow-body-style
 ): IGetServicePreferencesHandler => {
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
   return async (fiscalCode, serviceId) =>
     pipe(
-      profileModels.findLastVersionByModelId([fiscalCode]),
-      TE.mapLeft(failure =>
-        ResponseErrorQuery("Error while retrieving the profile", failure)
-      ),
-      TE.chainW(getProfileOrErrorResponse),
-      TE.filterOrElseW(nonLegacyServicePreferences, () =>
-        ResponseErrorConflict("Legacy service preferences not allowed")
+      sequenceS(TE.ApplicativeSeq)({
+        profile: getProfileOrErrorResponse(profileModel)(fiscalCode),
+        service: getServiceOrErrorResponse(serviceModel)(serviceId)
+      }),
+      TE.filterOrElseW(
+        ({ profile }) => nonLegacyServicePreferences(profile),
+        () => ResponseErrorConflict("Legacy service preferences not allowed")
```

Comment:
How will it work for not updated app ?

Reply:
In here the same as before. `LEGACY` profile cannot get `ServicePreferences`.
For `SPECIAL` services the service preferences are the same except for `isInboxEnabled` (the value of this depends only by the Activation status).